pynvjitlink/tests/test_pynvjitlink.py

```diff
@@ -74,14 +73,15 @@ def test_get_error_log(undefined_extern_cubin, gpu_arch_flag):
     filename, data = undefined_extern_cubin
     input_type = InputType.CUBIN.value
     _nvjitlinklib.add_data(handle, input_type, data, filename)
-    with pytest.raises(RuntimeError):
+    with pytest.raises(_nvjitlinklib.nvJitLinkError):
+        # FIXME: For some reason this API would leak the error log to stderr
         _nvjitlinklib.complete(handle)
     error_log = _nvjitlinklib.get_error_log(handle)
     _nvjitlinklib.destroy(handle)
     assert (
         "Undefined reference to '_Z5undefff' "
         "in 'undefined_extern.cubin'" in error_log
-    )
+    ), f"{error_log=}"
```

Comment:
This is the failing test:
```
$ pytest test_pynvjitlink.py  --maxfail=3
============================================================= test session starts =============================================================
platform linux -- Python 3.12.5, pytest-8.3.3, pluggy-1.5.0
rootdir: /local/home/leof/dev/pynvjitlink
configfile: pyproject.toml
collected 20 items                                                                                                                            

test_pynvjitlink.py ............F.......                                                                                                [100%]

================================================================== FAILURES ===================================================================
_____________________________________________________________ test_get_error_log ______________________________________________________________

undefined_extern_cubin = ('undefined_extern.cubin', b'\x7fELF\x02\x01\x013\x07\x00\x00\x00\x00\x00\x00\x00\x01\x00\xbe\x00~\x00\x00\x00\x00\x00...0\x00\x00\x00\x00\x00\x03\x00\x00\x00\x07\x00\x00\x18\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')
gpu_arch_flag = '-arch=sm_89'

    def test_get_error_log(undefined_extern_cubin, gpu_arch_flag):
        handle = _nvjitlinklib.create(gpu_arch_flag)
        filename, data = undefined_extern_cubin
        input_type = InputType.CUBIN.value
        _nvjitlinklib.add_data(handle, input_type, data, filename)
        with pytest.raises(_nvjitlinklib.nvJitLinkError):
            # FIXME: For some reason this API would leak the error log to stderr
            _nvjitlinklib.complete(handle)
        error_log = _nvjitlinklib.get_error_log(handle)
        _nvjitlinklib.destroy(handle)
>       assert (
            "Undefined reference to '_Z5undefff' "
            "in 'undefined_extern.cubin'" in error_log
        ), f"{error_log=}"
E       AssertionError: error_log='ERROR 9: finish\n\x00'
E       assert "Undefined reference to '_Z5undefff' in 'undefined_extern.cubin'" in 'ERROR 9: finish\n\x00'

test_pynvjitlink.py:81: AssertionError
------------------------------------------------------------ Captured stderr call -------------------------------------------------------------
error   : Undefined reference to '_Z5undefff' in '�����'
=========================================================== short test summary info ===========================================================
FAILED test_pynvjitlink.py::test_get_error_log - AssertionError: error_log='ERROR 9: finish\n\x00'
======================================================== 1 failed, 19 passed in 0.57s =========================================================
```

Reply:
This might be a regression in the version of nvJitLink used. Is that on your local system? What is the nvJitLink version?